GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java

```diff
@@ -104,223 +104,194 @@ public TaskServlet() {
      *
-     * @param fileName
-     * @param phoneNumber
-     * @param imei
-     * @param checksum
-     * @param offset
+     * @param fileProcessTaskRequest
      */
-    private ArrayList<SurveyInstance> processFile(
-            String fileName,
-            String phoneNumber,
-            String imei,
-            String checksum,
-            Integer offset) {
+    private ArrayList<SurveyInstance> processFile(TaskRequest fileProcessTaskRequest) {
+        String fileName = fileProcessTaskRequest.getFileName();
+        String phoneNumber = fileProcessTaskRequest.getPhoneNumber();
+        String imei = fileProcessTaskRequest.getImei();
+        String checksum = fileProcessTaskRequest.getChecksum();
+        Integer offset = fileProcessTaskRequest.getOffset();
+
         ArrayList<SurveyInstance> surveyInstances = new ArrayList<SurveyInstance>();
 
+        DeviceFilesDao dfDao = new DeviceFilesDao();
+        String url = DEVICE_FILE_PATH + fileName;
+
         try {
+            S3Util.putObjectAcl(BUCKET_NAME, OBJECTKEY_PREFIX + fileName, S3Util.ACL.PRIVATE);
+        } catch (IOException e) {
+            log.log(Level.SEVERE, "Error trying to secure zip file: " + e.getMessage(), e);
+        }
 
-            DeviceFilesDao dfDao = new DeviceFilesDao();
-            String url = DEVICE_FILE_PATH + fileName;
+        URLConnection conn = null;
+        BufferedInputStream deviceZipFileInputStream = null;
 
-            try {
-                S3Util.putObjectAcl(BUCKET_NAME, OBJECTKEY_PREFIX + fileName, S3Util.ACL.PRIVATE);
-            } catch (IOException e) {
-                log.log(Level.SEVERE, "Error trying to secure zip file: " + e.getMessage(), e);
-            }
+        try {
+            conn = S3Util.getConnection(BUCKET_NAME, OBJECTKEY_PREFIX + fileName);
+            deviceZipFileInputStream = new BufferedInputStream(conn.getInputStream());
+        } catch (IOException e) {
+            rescheduleTask(fileProcessTaskRequest);
+        }
+
+        List<DeviceFiles> dfList = null;
+        DeviceFiles deviceFile = null;
+        dfList = dfDao.listByUri(url);
+        if (dfList != null && dfList.size() > 0) {
+            deviceFile = dfList.get(0);
+        }
+        if (deviceFile == null) {
+            deviceFile = new DeviceFiles();
+        }
+        deviceFile.setProcessDate(getNowDateTimeFormatted());
+        deviceFile.setProcessedStatus(StatusCode.IN_PROGRESS);
+        deviceFile.setURI(url);
+        deviceFile.setPhoneNumber(phoneNumber);
+        deviceFile.setImei(imei);
+        deviceFile.setChecksum(checksum);
+        deviceFile.setUploadDateTime(new Date());
+        Date collectionDate = new Date();
+
+        ZipInputStream deviceFilesStream = new ZipInputStream(deviceZipFileInputStream);
+        ArrayList<String> unparsedLines = null;
+        try {
+            unparsedLines = extractDataFromZip(deviceFilesStream);
+        } catch (Exception iex) {
```

Comment:
Isn't this situation entitled to reschedule the task? If so, wouldn't it make sense to merge the previous try/catch block with this one? Any runtime exception will prevent this zip file from being reprocessed as far as I understand